### PR TITLE
Matching tiles game freezing issue fixed #1325

### LINKED ISSUE #1325: Matching tiles game freezes if home button pressed and app reopened
## Description
The matching tiles game freezes if the home button is pressed midway and again resumed. It causes infinite number of loops. 
To recreate the issue, press home button midway and then reopen the app. 

## Mocks
whatsappvideo20190124at61

### Update [Required]
- [ ] Ensure game resumes if home button pressed and app opened again.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
24 hours

### PR DESCRIPTION
**Pull request by apoorva-mk**
### Description
Fixed the cause of timer stopping when home button pressed and game resumed by adding onRestart() method.

Fixes # 1325

### Type of Change:
- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Tested it on various possible scenarios and the app performed as expected.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

20190129002125720x14400

### CODE REVIEW

**Review by Rimjhim28 — CHANGES_REQUESTED** [review]
@apoorva-mk The changes look good. Please rename the ```Restart``` variable to ```restart```. We follow camelcases for naming variables.

File: PowerUp/app/src/main/java/powerup/systers/com/memory_match_game/MemoryMatchGameActivity.java [hunk 1]
```diff
@@ -52,6 +52,9 @@
     public boolean calledFromActivity = true, correctAns = true, buttonClick = false;;
     private long millisLeft = 30000;
 
+    //Variable to see if
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/memory_match_game/MemoryMatchGameActivity.java]: The comment here looks incomplete

File: PowerUp/app/src/main/java/powerup/systers/com/memory_match_game/MemoryMatchGameActivity.java [hunk 2]
```diff
@@ -256,4 +260,12 @@ public void onBackPressed() {
         startActivity(new Intent(MemoryMatchGameActivity.this, MapActivity.class));
         overridePendingTransition(R.animator.fade_in_custom, R.animator.fade_out_custom);
     }
+
+    public void onRestart()
+    {
```
Comment (Rimjhim28) [inline PowerUp/app/src/main/java/powerup/systers/com/memory_match_game/MemoryMatchGameActivity.java]: follow proper parenthesis pattern.
it should have been:
```public void onRestart() { ```

**Review by Rimjhim28 — CHANGES_REQUESTED** [review]
Use an appropriate commit message.

**Comment by apoorva-mk** [comment]
Hey I've made the suggested changes! Please review it, thank you :)

**Review by Rimjhim28 — CHANGES_REQUESTED** [review]
@apoorva-mk Squash the commits.

**Pull request closed without merging**